src/core/core.controller.js

```diff
@@ -44,12 +44,21 @@ module.exports = function(Chart) {
 		var newOptions = chart.options;
 
 		// Update Scale(s) with options
-		if (newOptions.scale) {
-			chart.scale.options = newOptions.scale;
-		} else if (newOptions.scales) {
-			newOptions.scales.xAxes.concat(newOptions.scales.yAxes).forEach(function(scaleOptions) {
-				chart.scales[scaleOptions.id].options = scaleOptions;
-			});
+		if (newOptions.scale || newOptions.scales) {
+			newOptions = helpers.configMerge(
+				Chart.defaults.global,
+				Chart.defaults[chart.config.type],
+				newOptions || {});
```

Comment:
This is actually creating a new options object every update. A flag set by the setter of `chart.options.scales` can be used to skip unnecessary change. 
As the `update` function isn't called frequently I guess it's ok to update every time?